solutions/CF Santa's Bot.mdx

```diff
@@ -0,0 +1,152 @@
+---
+id: cf-santa-bot
+source: CF
+title: Santa's Bot
+author: Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/72577)
+
+Clarification:
+The optimization the author uses is this property of fractions modulo $998244353$ is this:
+
+Say we had two fractions $x=\frac{a}{b}$ and $y=\frac{c}{d}$. Let $f$ be a function that calculates the "total" of a fraction (that is, the numerator times the modular inverse of the denominator). To find $f(x + y)$, we can simply calculate $f(x) + f(y)$. Thus, to find the total of the overall probability, we can just add up all the totals of all the smaller probabilities, keeping a running sum as we go.
```

Comment:
i think it would be better to wrap this in an `<Info>` block

Reply:
What would that do?

Reply:
it would look better :P